clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -78,15 +86,24 @@ invariant_checker_t::parallel_shard_supported()
 }
 
 void *
-invariant_checker_t::parallel_shard_init(int shard_index, void *worker_data)
+invariant_checker_t::parallel_shard_init_stream(int shard_index, void *worker_data,
+                                                memtrace_stream_t *shard_stream)
 {
     auto per_shard = std::unique_ptr<per_shard_t>(new per_shard_t);
+    per_shard->stream = shard_stream;
     void *res = reinterpret_cast<void *>(per_shard.get());
     std::lock_guard<std::mutex> guard(shard_map_mutex_);
     shard_map_[shard_index] = std::move(per_shard);
     return res;
 }
 
+// We have no stream interface in invariant_checker_test unit tests.
```

Comment:
Instead of adding complexity here, can we just use the stream interface in invariant_checker_test?

Reply:
Do you have a suggestion for how to implement that?  To me it does not look easy.  There is no reader in that test; it uses synthetic sequences and custom process_memref calls.  A reader taking a vector?  Maybe in the end that would be nice; but it sure seems like a lot more work than the tweaks here.

Reply:
Added a comment about possibly refactoring the test.